frontend/test/pytest/test_peephole_optimizations.py

```diff
@@ -29,6 +29,229 @@
+        assert np.allclose(result, expected)
+
+    ### Test two consecutive PauliZ gates get canceled out ###
+    @pytest.mark.parametrize("theta", [42.42])
+    def test_PauliZ(self, theta, backend):
+
+        @qjit
+        @qml.qnode(qml.device(backend, wires=1))
+        def f(theta):
+            qml.RX(theta, wires=0)
+            return qml.probs()
+
+        @qjit
+        @cancel_inverses
+        @qml.qnode(qml.device(backend, wires=1))
+        def g(theta):
+            qml.RX(theta, wires=0)
+            qml.PauliZ(wires=0)
+            qml.PauliZ(wires=0)
+            return qml.probs()
+
+        expected = f(theta)
+        result = g(theta)
+        assert np.allclose(result, expected)
+
+    ### Test two consecutive CNOT gates get canceled out ###
+    @pytest.mark.parametrize("theta", [42.42])
+    def test_CNOT(self, theta, backend):
+
+        @qjit
+        @qml.qnode(qml.device(backend, wires=2))
+        def f(theta):
+            qml.RX(theta, wires=0)
+            qml.RX(theta, wires=1)
+            return qml.probs()
+
+        @qjit
+        @cancel_inverses
+        @qml.qnode(qml.device(backend, wires=2))
+        def g(theta):
+            qml.RX(theta, wires=0)
+            qml.RX(theta, wires=1)
+            qml.CNOT(wires=(0, 1))
+            qml.CNOT(wires=(0, 1))
+            return qml.probs()
+
+        expected = f(theta)
+        result = g(theta)
+        assert np.allclose(result, expected)
+
+    ### Test two consecutive CY gates get canceled out ###
+    @pytest.mark.parametrize("theta", [42.42])
+    def test_CY(self, theta, backend):
+
+        @qjit
+        @qml.qnode(qml.device(backend, wires=2))
+        def f(theta):
+            qml.RX(theta, wires=0)
+            qml.RX(theta, wires=1)
+            return qml.probs()
+
+        @qjit
+        @cancel_inverses
+        @qml.qnode(qml.device(backend, wires=2))
+        def g(theta):
+            qml.RX(theta, wires=0)
+            qml.RX(theta, wires=1)
+            qml.CY(wires=(0, 1))
+            qml.CY(wires=(0, 1))
+            return qml.probs()
+
+        expected = f(theta)
+        result = g(theta)
+        assert np.allclose(result, expected)
+
+    ### Test two consecutive CZ gates get canceled out ###
+    @pytest.mark.parametrize("theta", [42.42])
+    def test_CZ(self, theta, backend):
+
+        @qjit
+        @qml.qnode(qml.device(backend, wires=2))
+        def f(theta):
+            qml.RX(theta, wires=0)
+            qml.RX(theta, wires=1)
+            return qml.probs()
+
+        @qjit
+        @cancel_inverses
+        @qml.qnode(qml.device(backend, wires=2))
+        def g(theta):
+            qml.RX(theta, wires=0)
+            qml.RX(theta, wires=1)
+            qml.CZ(wires=(0, 1))
+            qml.CZ(wires=(0, 1))
+            return qml.probs()
+
+        expected = f(theta)
+        result = g(theta)
+        assert np.allclose(result, expected)
+
+    ### Test two consecutive SWAP gates get canceled out ###
```

Comment:
Can we add a test case for out-of-order multi-bit gates are not canceled? e.g. `CNOT(wires=[0,1])---CNOT(wires=[1,0])` should not be canceled

Reply:
same for lit test